### PR TITLE
[CRIMAPP-1355] Add decisions

### PR DESCRIPTION
**Pull request by hiboabd**
## Description of change
Updated the application complete route to accept an optional array of decisions created in Review, as well as a new `decisions` table to persist them.

## Link to relevant ticket
[CRIMAPP-1355](https://dsdmoj.atlassian.net/browse/CRIMAPP-1355)

[CRIMAPP-1355]: https://dsdmoj.atlassian.net/browse/CRIMAPP-1355?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/api/datastore/v1/reviewing.rb [hunk 1]
```diff
@@ -32,14 +32,21 @@ class Reviewing < Base
         desc 'Mark an application as complete.'
         params do
           requires :application_id, type: String, desc: 'Crime Application UUID'
+          optional :decisions, type: [JSON] do
+            requires :interests_of_justice, type: JSON
+            requires :funding_decision, type: String
+            requires :comment, type: String
```
Comment (timpeat) [inline app/api/datastore/v1/reviewing.rb:38]: Comment is optional. As will be means. Are you thinking of adding means later?

File: db/migrate/20241002110620_create_decisions.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class CreateDecisions < ActiveRecord::Migration[7.1]
+  def change
+    create_table :decisions, id: :uuid do |t|
+      t.references :crime_application, type: :uuid, foreign_key: true, null: true, index: { unique: true }
```
Comment (timpeat) [inline db/migrate/20241002110620_create_decisions.rb:4]: Why is crime_application `null: true` ?

**Review by timpeat — COMMENTED** [review]
Looking great! Couple of minor comments.

File: app/api/datastore/v1/reviewing.rb (hunk 1 above)
Comment (EdwinKruglov) [reply app/api/datastore/v1/reviewing.rb:38]: I'll update to make the comment optional. Yes, we were initially thinking of leaving means for later, but I'm happy to add it now as well.

File: db/migrate/20241002110620_create_decisions.rb (hunk 2 above)
Comment (hiboabd) [reply db/migrate/20241002110620_create_decisions.rb:4]: Copy and paste error :sweat_smile: my bad!

File: app/api/datastore/v1/reviewing.rb (hunk 1 above)
Comment (timpeat) [reply app/api/datastore/v1/reviewing.rb]: Thanks. (Fine for means to be done later.)

**Review by timpeat — APPROVED** [review]
👏

**Pull request merged**